app/modules/conversations/conversation/conversation_service.py

```
@@ -724,26 +498,57 @@ async def _generate_and_stream_ai_response(
         project_id = conversation.project_ids[0] if conversation.project_ids else None
-        supervisor = SimplifiedAgentSupervisor(self.sql_db, self.provider_service)
-        await supervisor.initialize(user_id)
+
+        try:
+            history = self.history_manager.get_session_history(user_id, conversation_id)
+            validated_history = [
+                (str(msg.content) if msg.content else msg) for msg in history
+            ]
+
+        except Exception:
+            raise ConversationServiceError("Failed to get chat history")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)